### PR TITLE
Allow non-numeric data in pivot cells

### LINKED ISSUE #8926: No data displayed when pivoting data
With Metabase 0.30.4 and before pivoting a table worked by selecting 3 columns, where 2 columns were displayed and the last one was used as the information column at the intersections of column 1 and 2. With version 0.31 this seems no longer be working, the appreciated column selector works, but no data is displayed on the intersections.

Running Metabase 0.31.0 within Docker connected to a PostgreSQL database.

Steps to reproduce:
* Create a question in table form, where 3 columns are selected for display
* Select pivot and the pivoting column in the table settings
* Metabase displays empty intersection where data should be

screenshot_pivot3

screenshot_pivot2

### PR DESCRIPTION
**Pull request by paulrosenzweig**
Resolves #8926 

I'm not sure if this also helps #10025. As far as I can tell that _should_ work, and I was unable to repro. 

---

This PR allows pivot tables to display any column in the cells. To accomplish this, I updated the hiding logic for the `cell_column` drop down, the options within the dropdown, how we get the default for that dropdown.

### CODE REVIEW

File: frontend/test/metabase/lib/data_grid.unit.spec.js [hunk 1]
```diff
@@ -36,6 +36,27 @@ describe("data_grid", () => {
...
+        "Dimension 1",
+        "x",
+        "y",
+        "z",
+      ]);
+      expect(pivotedData.rows.map(row => [...row])).toEqual([
+        ["a", "q", "w", "e"],
+        ["b", "r", "t", "y"],
+      ]);
+    });
```
Comment (paulrosenzweig) [inline frontend/test/metabase/lib/data_grid.unit.spec.js]: This test already passed without changes. I just thought there should be an assertion about non-numeric data around the core pivoting code.

File: frontend/src/metabase/visualizations/visualizations/Table.jsx [hunk 2]
```diff
@@ -130,10 +130,14 @@ export default class Table extends Component {
...
         ],
-        settings,
+        { "table.pivot": pivotCol },
       ) => {
-        const col = cols.filter(isMetric)[0];
-        return col && col.name;
+        // We try to show numeric values in pivot cells, but if none are
+        // available, we fall back to the last column in the unpivoted table
+        const nonPivotCols = cols.filter(c => c.name !== pivotCol);
+        const lastCol = nonPivotCols[nonPivotCols.length - 1];
```
Comment (paulrosenzweig) [inline frontend/src/metabase/visualizations/visualizations/Table.jsx]: I was iffy on my choice to default to the last column if there are no metrics available. My thinking was that it seems more common to have grouped columns as the first two.

File: frontend/src/metabase/visualizations/visualizations/Table.jsx [hunk 3]
```diff
@@ -130,10 +130,14 @@ export default class Table extends Component {
             data: { cols, rows },
           },
         ],
-        settings,
+        { "table.pivot": pivotCol },
```
Comment (tlrobinson) [inline frontend/src/metabase/visualizations/visualizations/Table.jsx]: `table.pivot` is a boolean, I think you want `table.pivot_column`.

We should probably have some tests of the settings themselves. There are a couple in `visualization.unit.spec.js`

**Review by tlrobinson — DISMISSED** [review]
Just one thing that doesn't look right, but it would be good to add a test to catch that if it is indeed broken.

**Comment by mazameli** [comment]
@tlrobinson Looks like @paulrosenzweig added a test + fix per your comments. Should we merge this now?

File: frontend/src/metabase/visualizations/visualizations/Table.jsx [hunk 4]
```diff
@@ -124,16 +124,13 @@ export default class Table extends Component {
...
-        settings,
-      ) => {
-        const col = cols.filter(isMetric)[0];
-        return col && col.name;
+      getDefault: ([{ data }], { "table.pivot_column": pivotCol }) => {
+        // We try to show numeric values in pivot cells, but if none are
+        // available, we fall back to the last column in the unpivoted table
+        const nonPivotCols = data.cols.filter(c => c.name !== pivotCol);
+        const lastCol = nonPivotCols[nonPivotCols.length - 1];
+        const [{ name } = lastCol] = nonPivotCols.filter(isMetric);
```
Comment (tlrobinson) [inline frontend/src/metabase/visualizations/visualizations/Table.jsx]: This destructuring kind of weirds me out, but not a big deal. I think

```
const { name } = nonPivotCols.find(isMetric) || lastCol;
```

would be a bit clearer.

File: frontend/test/metabase/visualizations/lib/settings/visualization.unit.spec.js [hunk 5]
```diff
@@ -59,29 +59,74 @@ describe("visualization_settings", () => {
         ));
     });
   });
+  describe("getDefault", () => {
```
Comment (tlrobinson) [inline frontend/test/metabase/visualizations/lib/settings/visualization.unit.spec.js]: Seems like 

```
describe("table.cell_column", () => {
  describe("getDefault", () => {
```

would make more sense (or just remove the `getDefault` describe entirely)

**Review by tlrobinson — APPROVED** [review]
Just a couple minor things that would be nice but not required

File: frontend/src/metabase/visualizations/visualizations/Table.jsx (hunk 4 above)
Comment (paulrosenzweig) [reply frontend/src/metabase/visualizations/visualizations/Table.jsx]: 👍 that's much better

**Pull request merged**